### ISSUE TITLE
Unique Constraint SYS_C00399566 violated

### ISSUE THREAD

**Issue by novumcastrum** [utPLSQL/utPLSQL#1239]
**Describe the bug**
When executing a test run with code coverage, the unique constraint violation SYS_C00399566 occurs.

I guess that a previous test run did not completely finish because the computer where utplsql-cli was called ran out of memory to save the reports. Now all further test runs run against this constraint.

Which code coverage tables need to be cleared? Is there a script for this? 

Error Trace:
```
########### utPLSQL cli ############
#                                  #
#   utPLSQL-cli 3.1.9.local        #
#   utPLSQL-java-api 3.1.9.local   #
#   Java-Version: 11.0.11          #
#   ORACLE_HOME: null              #
#   NLS_LANG: null                 #
#                                  #
#   Thanks for testing!            #
#                                  #
####################################

Use connection string jdbc:oracle:thin:****/****@****:****:****
Successfully connected to database. UtPLSQL core: v3.1.11.3559
Oracle-Version: 19.0.0.0.0
Running tests now.
--------------------------------------
TestRunner initialized
Running on utPLSQL v3.1.11.3559
Initializing reporters
Running tests
ORA-00001: Unique Constraint (UT_FRAMEWORK.SYS_C00399566) verletzt
ORA-06512: in "UT_FRAMEWORK.UT_RUNNER", Zeile 152
ORA-06512: in "UT_FRAMEWORK.UT_COVERAGE_HELPER", Zeile 89
ORA-06512: in "UT_FRAMEWORK.UT_COVERAGE", Zeile 183
ORA-06512: in "UT_FRAMEWORK.UT_COVERAGE_REPORTER_BASE", Zeile 22
ORA-06512: in "UT_FRAMEWORK.UT_REPORTER_BASE", Zeile 173
ORA-06512: in "UT_FRAMEWORK.UT_EVENT_MANAGER", Zeile 70
ORA-06512: in "UT_FRAMEWORK.UT_EVENT_MANAGER", Zeile 80
ORA-06512: in "UT_FRAMEWORK.UT_RUN", Zeile 52
ORA-06512: in "UT_FRAMEWORK.UT_SUITE_ITEM", Zeile 49
ORA-06512: in "UT_FRAMEWORK.UT_RUNNER", Zeile 144
ORA-06512: in Zeile 1

java.sql.SQLIntegrityConstraintViolationException: ORA-00001: Unique Constraint (UT_FRAMEWORK.SYS_C00399566) verletzt
ORA-06512: in "UT_FRAMEWORK.UT_RUNNER", Zeile 152
ORA-06512: in "UT_FRAMEWORK.UT_COVERAGE_HELPER", Zeile 89
ORA-06512: in "UT_FRAMEWORK.UT_COVERAGE", Zeile 183
ORA-06512: in "UT_FRAMEWORK.UT_COVERAGE_REPORTER_BASE", Zeile 22
ORA-06512: in "UT_FRAMEWORK.UT_REPORTER_BASE", Zeile 173
ORA-06512: in "UT_FRAMEWORK.UT_EVENT_MANAGER", Zeile 70
ORA-06512: in "UT_FRAMEWORK.UT_EVENT_MANAGER", Zeile 80
ORA-06512: in "UT_FRAMEWORK.UT_RUN", Zeile 52
ORA-06512: in "UT_FRAMEWORK.UT_SUITE_ITEM", Zeile 49
ORA-06512: in "UT_FRAMEWORK.UT_RUNNER", Zeile 144
ORA-06512: in Zeile 1

        at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:509)
        at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:461)
        at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1104)
        at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:550)
        at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:268)
        at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:655)
        at oracle.jdbc.driver.T4CCallableStatement.doOall8(T4CCallableStatement.java:265)
        at oracle.jdbc.driver.T4CCallableStatement.doOall8(T4CCallableStatement.java:86)
        at oracle.jdbc.driver.T4CCallableStatement.executeForRows(T4CCallableStatement.java:965)
        at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1205)
        at oracle.jdbc.driver.OraclePreparedStatement.executeInternal(OraclePreparedStatement.java:3666)
        at oracle.jdbc.driver.T4CCallableStatement.executeInternal(T4CCallableStatement.java:1358)
        at oracle.jdbc.driver.OraclePreparedStatement.execute(OraclePreparedStatement.java:3778)
        at oracle.jdbc.driver.OracleCallableStatement.execute(OracleCallableStatement.java:4251)
        at oracle.jdbc.driver.OraclePreparedStatementWrapper.execute(OraclePreparedStatementWrapper.java:1081)
        at org.utplsql.api.testRunner.DynamicTestRunnerStatement.execute(DynamicTestRunnerStatement.java:84)
        at org.utplsql.api.TestRunner.run(TestRunner.java:223)
        at org.utplsql.cli.RunTestRunnerTask.call(RunTestRunnerTask.java:45)
        at org.utplsql.cli.RunTestRunnerTask.call(RunTestRunnerTask.java:24)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: Error : 1, Position : 0, Sql = BEGIN ut_runner.run(a_paths => :1 , a_reporters => :2 , a_color_console => (case :3  when 1 then true else false end), a_fail_on_errors => (case :4  when 1 then true else false end), a_client_character_set => :5 , a_random_test_order => (case :6  when 1 then true else false end));END;, OriginalSql = BEGIN ut_runner.run(a_paths => ?, a_reporters => ?, a_color_console => (case ? when 1 then true else false end), a_fail_on_errors => (case ? when 1 then true else false end), a_client_character_set => ?, a_random_test_order => (case ? when 1 then true else false end));END;, Error Msg = ORA-00001: Unique Constraint (UT_FRAMEWORK.SYS_C00399566) verletzt
ORA-06512: in "UT_FRAMEWORK.UT_RUNNER", Zeile 152
ORA-06512: in "UT_FRAMEWORK.UT_COVERAGE_HELPER", Zeile 89
ORA-06512: in "UT_FRAMEWORK.UT_COVERAGE", Zeile 183
ORA-06512: in "UT_FRAMEWORK.UT_COVERAGE_REPORTER_BASE", Zeile 22
ORA-06512: in "UT_FRAMEWORK.UT_REPORTER_BASE", Zeile 173
ORA-06512: in "UT_FRAMEWORK.UT_EVENT_MANAGER", Zeile 70
ORA-06512: in "UT_FRAMEWORK.UT_EVENT_MANAGER", Zeile 80
ORA-06512: in "UT_FRAMEWORK.UT_RUN", Zeile 52
ORA-06512: in "UT_FRAMEWORK.UT_SUITE_ITEM", Zeile 49
ORA-06512: in "UT_FRAMEWORK.UT_RUNNER", Zeile 144
ORA-06512: in Zeile 1

        at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:513)
        ... 22 more
```

**Provide version info**
utplsql v3.1.11.3559
Oracle Database 19c Enterprise Edition Release 19.0.0.0.0 - Production

NLS_LANGUAGE	AMERICAN
NLS_TERRITORY	GERMANY
NLS_CURRENCY	€
NLS_ISO_CURRENCY	GERMANY
NLS_NUMERIC_CHARACTERS	,.
NLS_CALENDAR	GREGORIAN
NLS_DATE_FORMAT	DD.MM.RR
NLS_DATE_LANGUAGE	AMERICAN
NLS_SORT	BINARY
NLS_TIME_FORMAT	HH24:MI:SSXFF
NLS_TIMESTAMP_FORMAT	DD.MM.RR HH24:MI:SSXFF
NLS_TIME_TZ_FORMAT	HH24:MI:SSXFF TZR
NLS_TIMESTAMP_TZ_FORMAT	DD.MM.RR HH24:MI:SSXFF TZR
NLS_DUAL_CURRENCY	€
NLS_COMP	BINARY
NLS_LENGTH_SEMANTICS	BYTE
NLS_NCHAR_CONV_EXCP	FALSE

**Information about client software**
utplsql-cli, TOAD (same error)

**To Reproduce**
e.g. with utplsql-cli utplsql run ****/****@****:****:**** -p=xyz -f=ut_documentation_reporter -o=run.log -s -f=ut_coverage_sonar_reporter -o=sonarcover.xml -f=ut_sonar_test_reporter -o=sonartest.xml  -f=ut_junit_reporter -o=junit.xml 

**Expected behavior**
Bug should no longer occur.

**Comment by jgebal** [comment]
Hello @novumcastrum 
This is not an expected behaviour and I'm quite surprised to see this exception.
The exception is raised in insert into `ut_coverage_runs` table.
There are two unique constraints on that table but in normal use-cases none of them should be violated as each new utPLSQL run will generate a new `line_coverage_id` for data from `DBMS_PROFILER` as well as new `block_coverage_id` for data from `DBMS_PLSQL_CODE_COVERAGE`.

Now. The fact that you see this exception indicates that there is some problem. Maybe the sequence number are somehow reset in your system?

You can check output of the below query.
```sql
select column_name, position
  from dba_cons_columns 
 where table_name = 'UT_COVERAGE_RUNS' 
   and constraint_name = 'SYS_C00399566' 
   and owner = 'UT_FRAMEWORK';
```
This will show the column name for that constraint.
It will be one of the columns I've mentioned above.

Then you can check for values in that table for that column.
You can call for new coverage run to see what seq value you get for it:
```sql
set serverout on
begin
  dbms_output.put_line('value of new block_coverage_id= '
    || dbms_plsql_code_coverage.start_coverage(run_comment => 'test') );
  dbms_plsql_code_coverage.stop_coverage();
end;
/
begin
  dbms_output.put_line('value of new line_coverage_id= '
    || ut_coverage_runs.plsql_profiler_runnumber.nextval );
end;
/
```
 
As far as I can imagine, this error could only happen if the profiler or coverage sequences were somehow reset.

Clearing the table data should help.

I strongly suggest update to utPLSQL v 3.1.12 as there are some small but valuable improvements to coverage gathering.
Have a look at [3.1.12 release description](https://github.com/utPLSQL/utPLSQL/releases/tag/v3.1.12)

**Comment by novumcastrum** [comment]
Hello @jgebal, 
thank you for your quick and detailed reply. We decided to update to utPLSQL v 3.1.12.
I have already tried clearing the table ``ut_coverage_runs`` and there was no more error message, but the process did not terminate. Even for very small packages to be analyzed. I did not analyze further why the process did not end.

I leave the ticket open as long as we haven't finished the update and tested code coverage.
Thank you very much for your efforts!
